### PR TITLE
Admin UI for shield

### PR DESCRIPTION
**Pull request by pyadav**

### CODE REVIEW

File: internal/server/server.go [hunk 1]
```diff
@@ -90,8 +92,16 @@ func Serve(
 	if err != nil {
 		return err
 	}
+
 	httpMux.Handle("/jwks.json", jwksHandler)
 
+	spaHandler, err := spa.Handler(ui.Assets, "dist/ui", "index.html", false)
+	if err != nil {
+		fmt.Println("Failed to load spa:", err)
```
Comment (kushsharma) [inline internal/server/server.go:100]: Can we use an existing logger struct with `warn` tag?

File: ui/Makefile [hunk 2]
```diff
@@ -0,0 +1,10 @@
+.PHONY: dist test
+
+dist:
```
Comment (kushsharma) [inline ui/Makefile:3]: Should we also add a small README.md in `ui` directory? Listing what steps to run for running/building ui?

File: Makefile [hunk 3]
```diff
@@ -1,15 +1,19 @@
 GOVERSION := $(shell go version | cut -d ' ' -f 3 | cut -d '.' -f 2)
 
-.PHONY: build check fmt lint test test-race vet test-cover-html help install proto
+.PHONY: build check fmt lint test test-race vet test-cover-html help install proto ui
 .DEFAULT_GOAL := build
 PROTON_COMMIT := "056db0eef064bc5f82343e6a5a312567a39d9ec6"
 
+ui:
```
Comment (kushsharma) [inline Makefile:7]: We should also ignore `ui` folder from various `test` commands

File: ui/vite.config.ts [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+
+// https://vitejs.dev/config/
+export default defineConfig({
+  build: {
+    outDir: "dist/ui",
+  },
+  server: {
+    proxy: {
+      "/admin": {
+        target: "http://localhost:8000",
```
Comment (kushsharma) [inline ui/vite.config.ts:13]: Why is this hardcoded? Is it possible to put it in some .env?

File: internal/server/server.go (hunk 1 above)
Comment (pyadav) [reply internal/server/server.go]: added `logger.Warn`

File: ui/Makefile (hunk 2 above)
Comment (pyadav) [reply ui/Makefile:3]: Added

File: Makefile (hunk 3 above)
Comment (pyadav) [reply Makefile:7]: just added into make test, rest of them are using specific folder to run test

File: ui/vite.config.ts (hunk 4 above)
Comment (pyadav) [reply ui/vite.config.ts]: Fixed

**Review by kushsharma — APPROVED** [review]

**Pull request merged**